lib/response-types/token-response-type.ts

```diff
@@ -0,0 +1,97 @@
+import { InvalidArgumentError } from '../errors';
+import { ImplicitGrantType } from '../grant-types';
+import { Client, Model, User } from '../interfaces';
+import { Request } from '../request';
+
+export class TokenResponseType {
+  accessToken: string;
+  accessTokenLifetime: number;
+  model: Model;
+  constructor(options: any = {}) {
+    if (!options.accessTokenLifetime) {
+      throw new InvalidArgumentError(
+        'Missing parameter: `accessTokenLifetime`',
+      );
+    }
+
+    this.accessToken = undefined;
+    this.accessTokenLifetime = options.accessTokenLifetime;
+    this.model = options.model;
+  }
+
+  /**
+   * Handle token response type.
+   */
+
+  async handle(
+    request: Request,
+    client: Client,
+    user: User,
+    uri: string,
+    scope: string,
+  ) {
+    if (!request) {
+      throw new InvalidArgumentError('Missing parameter: `request`');
+    }
+
+    if (!client) {
+      throw new InvalidArgumentError('Missing parameter: `client`');
+    }
+
+    const accessTokenLifetime = this.getAccessTokenLifetime(client);
+
+    const options = {
+      user,
+      scope,
+      model: this.model,
+      accessTokenLifetime,
+    };
+
+    const grantType = new ImplicitGrantType(options);
```

Comment:
why just ImplicitGrantType ImplicitGrantType?
why not PasswordGrantType,RefreshTokenGrantType,ClientCredentialsGrantType?